srcts/src/utils/index.ts

```diff
@@ -226,21 +229,24 @@ function mergeSort<T>(
 }
 
 // Escape jQuery selector metacharacters: !"#$%&'()*+,./:;<=>?@[\]^`{|}~
-const $escape = function (val: string): string {
+function $escape(val: undefined): undefined;
+function $escape(val: string): string;
+function $escape(val: string | undefined): string | undefined {
+  if (typeof val === "undefined") return val;
   return val.replace(/([!"#$%&'()*+,./:;<=>?@[\\\]^`{|}~])/g, "\\$1");
-};
+}
 
 // Maps a function over an object, preserving keys. Like the mapValues
 // function from lodash.
-function mapValues<V, R>(
-  obj: { [key: string]: V },
-  f: (value: V, key: string, obj: { [key: string]: V }) => R
-): { [key: string]: R } {
-  const newObj: { [key: string]: R } = {};
-
-  for (const key in obj) {
-    if (hasOwnProperty(obj, key)) newObj[key] = f(obj[key], key, obj);
-  }
+function mapValues<T extends { [key: string]: any }, R extends any>(
+  obj: T,
+  f: (value: MapValuesUnion<T>, key: string, object: typeof obj) => R
+): MapWithResult<T, R> {
+  const newObj = {} as MapWithResult<T, R>;
+
+  Object.keys(obj).forEach((key: keyof typeof obj) => {
+    newObj[key] = f(obj[key], key as string, obj);
+  });
```

Comment:
Using `Object.keys(obj).forEach()` to iterate over the object's keys/values. While the original code works, there is minimal typing guarantee.